src/hotspot/share/nmt/vmatree.cpp

```diff
@@ -206,12 +207,13 @@ VMATree::SummaryDiff VMATree::register_mapping(position A, position B, StateType
   }
 
   // Finally, we can register the new region [A, B)'s summary data.
-  SingleDiff& rescom = diff.tag[NMTUtil::tag_to_index(metadata.mem_tag)];
+  MemTag tag_to_change = use_tag_inplace ? stA.out.mem_tag() : metadata.mem_tag;
+  SingleDiff& rescom = diff.tag[NMTUtil::tag_to_index(tag_to_change)];
```

Comment:
This seems to be a bug fix to 8335091. You should open a separate mainline PR with this fix and add a testcase for it. Your fix should be integrated before this PR is.

Reply:
Moved to this PR: https://github.com/openjdk/jdk/pull/23771